component/otelcol/processor/filter/filter.go

```diff
@@ -0,0 +1,165 @@
+package filter
+
+import (
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/processor"
+	otel_service "github.com/grafana/agent/service/otel"
+	"github.com/mitchellh/mapstructure"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:          "otelcol.processor.filter",
+		Args:          Arguments{},
+		Exports:       otelcol.ConsumerExports{},
+		NeedsServices: []string{otel_service.ServiceName},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := filterprocessor.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+type traceConfig struct {
+	Span      []string `river:"span,attr,optional"`
+	SpanEvent []string `river:"spanevent,attr,optional"`
+}
+
+type metricConfig struct {
+	Metric    []string `river:"metric,attr,optional"`
+	Datapoint []string `river:"datapoint,attr,optional"`
+}
+type logConfig struct {
```

Comment:
Would you be able to also add the `include` and `exclude` properties please?

Reply:
As we [agreed](https://github.com/grafana/agent/pull/5562#issuecomment-1782042834), there is no need to include these.